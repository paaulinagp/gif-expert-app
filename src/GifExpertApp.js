import React, {useState} from 'react';
import AddCategory from './AddCaterory';

const GifExpertApp = () => {

  // const categories = ['Harry Potter', 'Hermione Granger', 'Ron Weasley'];

  const [categories, setCategories] = useState(['Harry Potter', 'Hermione Granger', 'Ron Weasley']);

  // const handleAdd = () => {
  //   // setCategories([...categories, 'Prueba'])
  //   setCategories(cats => [...cats, 'Prueba'])
  // };

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr/>
      
      <ol>
        {
          categories.map( (category) => {
            return (<li key={ category }>{ category} </li>)
          })
        }
      </ol>
    </>
  );
};

export default GifExpertApp;