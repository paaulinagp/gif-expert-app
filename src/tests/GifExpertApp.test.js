import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';
import '@testing-library/jest-dom';

describe('Pruebas del componente <GifExpertApp />', () => {
  test('debe cargar correctamente el componente', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar un arreglo de categorias', () => {
    const categories= ['Harry Potter', 'Hermione', 'Ron'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
