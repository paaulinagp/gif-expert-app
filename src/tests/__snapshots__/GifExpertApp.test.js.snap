// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> debe cargar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> debe de mostrar un arreglo de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Harry Potter"
      key="Harry Potter"
    />
    <GifGrid
      category="Hermione"
      key="Hermione"
    />
    <GifGrid
      category="Ron"
      key="Ron"
    />
  </ol>
</Fragment>
`;
